### PR TITLE
fixed some file

### PR DESCRIPTION
**Pull request by miss85246**
1. `.gitignore` add package-lock.json
2. Unified subtitle level, set to h4 in basics/middleware.md
3. fixed spell error: endpoing -> endpoint in basics/routing.md

### CODE REVIEW

**Comment by miss85246** [comment]
@ahopkins  In the process of translation, I found several places that need to be modified, and I made simple modifications.

**Comment by ahopkins** [comment]
Yes, there is still lots of work to be done. The goal is to remove that BETA notice at the bottom simultaneously with the release of v21.3 which will include a [PR to remove the docs from RTD](https://github.com/sanic-org/sanic/pull/2006).

File: src/guide/basics/middleware.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ async def halt_response(request, response):
 ```
 :---
 
-## Order of execution
+#### Order of execution
```
Comment (ahopkins) [inline src/guide/basics/middleware.md:107]: Why this and not H2?

Comment (miss85246) [reply src/guide/basics/middleware.md:107]: look at this file : [basics/headers.md](https://github.com/sanic-org/sanic-guide/blob/master/src/guide/basics/headers.md) `Tokens` and `Proxy headers` are part of `Headers`  and they are H4 title. 
In like manner, this also should be H4 title

**Pull request closed without merging**